nautilus_trader/data/aggregation.pyx

```diff
@@ -765,90 +821,77 @@ cdef class TimeBarAggregator(BarAggregator):
 
             start_time -= pd.Timedelta(seconds=step)
-        elif aggregation == BarAggregation.MINUTE:
-            start_time = now.floor(freq="h")
 
-            if self._time_bars_origin is not None:
-                start_time += self._time_bars_origin
+        elif aggregation == BarAggregation.MINUTE:
+            start_time = now.floor(freq="h") + self._time_bars_origin
 
             if now < start_time:
                 start_time -= pd.Timedelta(hours=1)
 
-            while start_time <= now:
+            while start_time < now:
                 start_time += pd.Timedelta(minutes=step)
 
             start_time -= pd.Timedelta(minutes=step)
-        elif aggregation == BarAggregation.HOUR:
-            start_time = now.floor(freq="d")
 
-            if self._time_bars_origin is not None:
-                start_time += self._time_bars_origin
+        elif aggregation == BarAggregation.HOUR:
+            start_time = now.floor(freq="d") + self._time_bars_origin
 
             if now < start_time:
                 start_time -= pd.Timedelta(days=1)
 
-            while start_time <= now:
+            while start_time < now:
                 start_time += pd.Timedelta(hours=step)
 
             start_time -= pd.Timedelta(hours=step)
-        elif aggregation == BarAggregation.DAY:
-            start_time = now.floor(freq="d")
 
-            if self._time_bars_origin is not None:
-                start_time += self._time_bars_origin
+        elif aggregation == BarAggregation.DAY:
+            start_time = now.floor(freq="d") + self._time_bars_origin
 
-            if now < start_time:
+            if now <= start_time:
                 start_time -= pd.Timedelta(days=1)
-        elif aggregation == BarAggregation.WEEK:
-            start_time = (now - pd.Timedelta(days=now.dayofweek)).floor(freq="d")
 
-            if self._time_bars_origin is not None:
-                start_time += self._time_bars_origin
+        elif aggregation == BarAggregation.WEEK:
+            start_time = (now - pd.Timedelta(days=now.dayofweek)).floor(freq="d") + self._time_bars_origin
 
-            if now < start_time:
+            if now <= start_time:
                 start_time -= pd.Timedelta(weeks=1)
-        elif aggregation == BarAggregation.MONTH:
-            start_time = (now - pd.DateOffset(months=now.month - 1, days=now.day - 1)).floor(freq="d")
 
-            if self._time_bars_origin is not None:
-                start_time += self._time_bars_origin
+        elif aggregation == BarAggregation.MONTH:
+            start_time = ((now - pd.DateOffset(months=now.month - 1, days=now.day - 1)).floor(freq="d")
+                          + self._time_bars_origin)
 
             if now < start_time:
                 start_time -= pd.DateOffset(years=1)
 
-            while start_time <= now:
+            while start_time < now:
                 start_time += pd.DateOffset(months=step)
 
             start_time -= pd.DateOffset(months=step)
+
```

Comment:
Lets not add a blank between keywords in an `if-elif-else` block, I think this makes the code less cohesive and harder to reason about.